pbft signing threshold config from CLI

avieth opened the pull request:
use the --pbft-threshold parameter

avieth committed:
pbft signing threshold config from CLI

use the --pbft-threshold parameter

avieth commented on src/exec/Main.hs:
```diff
@@ -222,6 +227,16 @@ cliParser = ByronProxyOptions
+    Opt.long "pbft-threshold" <>
+    Opt.metavar "DOUBLE" <>
+    -- FIXME ouroboros-consensus should export defaultPBftSignatureThreshold
+    -- That'd be better than putting this magic number here.
```
Alternatively, I could make this an optional parameter with no default, and give `Nothing` to `protocolInfoByron`. Any strong arguments in either favour?

dcoutts replied:
You can make this a `Maybe` field, since `protocolInfoByron` accepts a maybe, and it uses the default if you pass nothing. That'd be the simplest.

dcoutts commented on src/exec/Main.hs:
```diff
@@ -507,7 +522,7 @@ main = do
               protocolInfo = protocolInfoByron
                 newGenesisConfig
-                Nothing -- Default signature threshold.
+                (Just (bpoPBftSignatureThreshold bpo))
```
Pass the maybe though here, so it'll use the default in the nothing case.

avieth commented:
Agreed that having an optional parameter is better than a default value here. Only downside is that the default remains obscure (it's not even exported).

avieth committed:
make pbft threshold optional

Rather than have a default in CLI
The default is still there, by way of byronProtocolInfo from
ouroboros-consensus.

dcoutts reviewed (approved):
Perfect.

avieth commented:
bors r+
